### ISSUE TITLE
Add ability to see which channel an invoice was paid through

### ISSUE THREAD

**Issue by ashleyholman** [lightningnetwork/lnd#3169]
When one of my invoices is paid, I would like to be able to find out which channel it was paid through.  The lookupinvoice API doesn't appear to show this information.  It would be great if it had a parameter like "paid_via_channel" or something, to show that information.

The only record I found was 1 line in my log file, and I'm not sure how to convert that into a channel ID.

```
Jun 06 20:10:27 lightning lnd[9736]: 2019-06-06 20:10:27.206 [INF] HSWC: ChannelLink(579257:1314:0) settling htlc 5a01a8d122f1afbc5071901ff9724d467b48d0b32af64cc8fa14c0a434507fad as exit hop
```

**Comment by cfromknecht** [comment]
@ashleyholman this is indeed useful, but requires some pretty low-level changes to lnd internals. there is an ongoing refactor that will eventually enable this. the bulk of the work will be addressed in fixing https://github.com/lightningnetwork/lnd/issues/2208, then the final step would be exposing this info over rpc

**Comment by alexbosworth** [comment]
A workaround could be to use the hold invoice feature, but you'd have to find the matching held HTLC in listchannels and do the cross referencing yourself

**Comment by joostjager** [comment]
Fix coming up in #3390
